compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/util/DeepCopyIrTreeWithSymbols.kt

```diff
@@ -65,6 +65,14 @@ open class DeepCopyIrTreeWithSymbols(
     }
 
     protected fun mapDeclarationOrigin(origin: IrDeclarationOrigin) = origin
+    protected fun mapDeclarationOrigin(declaration: IrDeclaration):IrDeclarationOrigin {
```

Comment:
Won't it be enough to override this method in the inliner copier?

Reply:
will try